src/main/java/mate/academy/service/order/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+
+    @Transactional
+    @Override
+    public List<OrderDto> getAllOrder() {
+        User user = userService.getAuthenticatedUser();
+        List<Order> orders = orderRepository.getOrderByUserId(user.getId());
+        for (Order order : orders) {
+            order.setOrderItems(orderItemRepository.getAllByOrderId(order.getId()));
+        }
+        return orders.stream()
+                .map(this::initializeOrderItems)
+                .toList();
+    }
+
+    @Transactional
+    @Override
+    public OrderDto addAddress(OrderAddressDto orderAddressDto) {
+        Order order = createOrder(orderAddressDto);
+        order.setShippingAddress(orderAddressDto.getShippingAddress());
+        return orderMapper.toDto(order);
+    }
+
+    @Transactional
+    @Override
+    public OrderDto updateStatus(Long id, OrderStatusDto orderStatusDto) {
+        Order order = orderRepository.findOrderById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order with id " + id));
+        order.setStatus(orderStatusDto.status());
+        return initializeOrderItems(orderRepository.save(order));
+    }
+
+    @Transactional(readOnly = true)
+    @Override
+    public Set<OrderItemDto> getAllOrderItems(Long orderId) {
+        return getAllOrderItemByOrderId(orderId)
+                .collect(Collectors.toSet());
+    }
+
+    @Transactional(readOnly = true)
+    @Override
+    public OrderItemDto findByIdOrderItem(Long orderId, Long itemId) {
+        return getAllOrderItemByOrderId(orderId)
+                .filter(i -> i.getId().equals(itemId))
+                .findFirst().orElseThrow(
+                        () -> new EntityNotFoundException("Can't find order item by id "
+                                + itemId));
+    }
+
+    private Order createOrder(OrderAddressDto requestAddressDto) {
+        Order order = new Order();
+        ShoppingCart shoppingCart = shoppingCartService.getCurrentUserCart();
+        order.setOrderDate(LocalDateTime.now());
+        order.setStatus(Order.Status.PENDING);
+        order.setShippingAddress(requestAddressDto.getShippingAddress());
+        order.setUser(userService.getAuthenticatedUser());
+        order.setTotal(shoppingCart.getCartItems().stream()
+                .map(this::calculateItemPrice)
+                .reduce(BigDecimal::add)
+                .orElseThrow());
+        Order savedOrder = orderRepository.save(order);
+        order.setOrderItems(shoppingCart.getCartItems().stream()
+                .map(i -> orderItemMapper.cartItemToOrderItem(i, savedOrder))
+                .map(orderItemRepository::save)
+                .collect(Collectors.toSet()));
+        return savedOrder;
+    }
+
+    private Stream<OrderItemDto> getAllOrderItemByOrderId(Long orderId) {
+        return orderRepository.getOrderByUserId(userService.getAuthenticatedUser().getId())
+                .stream()
+                .filter(o -> o.getId().equals(orderId))
```

Comment:
Never use 1-2 letter abbreviations for naming, even in steams and lambdas.
In a week when you forget code context, you will hate such a decision.
Please recheck all methods
```suggestion
                .filter(order -> order.getId().equals(orderId))
```